### PR TITLE
Bach BQ: Remove Nano-precision from aggregated SeriesTimedelta

### PR DESCRIPTION
**Pull request by KathiaBarahona**
BigQuery supports till microseconds precision, when trying to use `dt` accessor from a SeriesTimedelta with nano-precision, this will cause an error when executing the query. Therefore, we must be always sure that the timedelta supports till microsecond precision.

~I'm still missing functional tests in bach,~ but seems that session_duration from modelhub is now generating same results for both PG and BQ

### CODE REVIEW

File: bach/bach/series/series_datetime.py [hunk 1]
```diff
@@ -620,7 +626,12 @@ def sum(self, partition: WrappedPartition = None,
             skipna=skipna,
             min_count=min_count
         )
-        return cast('SeriesTimedelta', result)
+        result = result.copy_override_type(SeriesTimedelta)
+
+        if is_bigquery(self.engine):
+            result = self._remove_nano_precision_bigquery(result)
```
Comment (KathiaBarahona) [inline bach/bach/series/series_datetime.py:632]: Probably is not necessary for `sum`. Is there any chance we result with nanoprecision here??

**Review by thijs-obj — APPROVED** [review]
One comment, sort of subjective, so fully up to you.

Looks good to me. Good solution.

**Question**: Have we filed a bug report with Google about this?

File: bach/bach/series/series_datetime.py [hunk 2]
```diff
@@ -631,7 +639,53 @@ def mean(self, partition: WrappedPartition = None, skipna: bool = True) -> 'Seri
...
+        # Format specifier %06.0f will format fractional part of seconds with maximum width of 6 digits
+        # for example:
+        # nanoseconds = 1142857, converting them into microseconds is 1142.857
+        # when applying string formatting, the value will be rounded into 1143 (.0 precision)
+        # and will be left padded by 2 leading zeros: 001143 (0 flag and 6 minimum width)
+        # for more information:
+        # https://cloud.google.com/bigquery/docs/reference/standard-sql/string_functions#format_string
+        format_expr = Expression.construct(
+            f'format({{}}, {{}})',
+            Expression.string_value(self._BQ_INTERVAL_FORMAT),
```
[the remark this answers is not in the document]
Comment (KathiaBarahona) [reply bach/bach/series/series_datetime.py:681]: yeah, thought the same

**Review by hendrik-obj — APPROVED** [review]
lgtm!

**Pull request merged**